graphdatascience/query_runner/protocol/project_protocols.py

```diff
@@ -132,15 +132,21 @@ def is_not_done(result: DataFrame) -> bool:
 
         logger = getLogger()
 
+        member_address = query_runner.call_procedure(
+            ProtocolVersion.V3.versioned_procedure_name(endpoint), params, yields, database, logging, False
+        ).squeeze()["host"]
+
+        projection_query_runner = query_runner.clone(f"neo4j+s://{member_address}")
```

Comment:
might want to do the same for write-back to make sure we dont get hit by a leader switch and start writing back a second time?

Reply:
The write backs are protected by the procedure being a write procedure. We generally do not survive a leaderswitch right now.